nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -181,23 +181,38 @@ public IcebergTable table(TableIdentifier tableIdentifier) {
   }
 
   public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    getRef().checkMutable();
+
+    if (namespace.isEmpty()) {
+      throw new IllegalArgumentException("Creating empty namespaces is not supported");
+    }
+
     try {
-      getRef().checkMutable();
-      withReference(
-              getApi()
-                  .createNamespace()
-                  .namespace(org.projectnessie.model.Namespace.of(namespace.levels()))
-                  .properties(metadata))
-          .create();
+      org.projectnessie.model.Namespace content =
+          org.projectnessie.model.Namespace.of(metadata, namespace.levels());
+      ContentKey key = content.toContentKey();
+
+      api.commitMultipleOperations()
+          .branch((Branch) getRef().getReference())
+          .commitMeta(NessieUtil.buildCommitMetadata("create namespace " + key, catalogOptions))
+          .operation(Operation.Put.of(key, content))
+          .commit();
```

Comment:
we should also add a retry like `renameTable()` and `dropTable()`, only for the `commitTable()` retry is not needed as Iceberg has a built in retry for it from a transaction.

Reply:
OK will do – but I think that most of the errors are not transient, so retrying won't really help.